src/app-layer-expectation.c

```diff
@@ -157,11 +181,30 @@ static Expectation *AppLayerExpectationLookup(Flow *f, int direction, IPPair **i
     return IPPairGetStorageById(*ipp, g_expectation_id);
 }
 
+
+static ExpectationList *AppLayerExpectationRemove(IPPair *ipp,
+                                                  ExpectationList *exp_list,
+                                                  Expectation *exp)
+{
+    CIRCLEQ_REMOVE(exp_list, exp, entries);
+    AppLayerFreeExpectation(exp);
+    SC_ATOMIC_SUB(expectation_count, 1);
+    exp_list->length--;
+    if (exp_list->length <= 0) {
```

Comment:
`length` is `uint8_t`, so `< 0` check is useless.

Reply:
Good catch.